documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1096 @@
+}
+----
+
+
+[[query_continuous]]
+== Continuous Query
+
+Continuous Queries allow an application to register a listener which will receive the entries that currently match a
+query filter, and will be continuously notified of any changes to the queried data set that result from further cache
+operations. This includes incoming matches, for values that have joined the set, updated matches, for matching values
+that were modified and continue to match, and outgoing matches, for values that have left the set. By using a Continuous
+Query the application receives a steady stream of events instead of having to repeatedly execute the same query to
+discover changes, resulting in a more efficient use of resources. For instance, all of the following use cases could
+utilize Continuous Queries:
+
+* Return all persons with an age between 18 and 25 (assuming the Person entity has an _age_ property and is updated by
+the user application).
+* Return all transactions higher than $2000.
+* Return all times where the lap speed of F1 racers were less than 1:45.00s (assuming the cache contains Lap entries and
+that laps are entered live during the race).
+
+=== Continuous Query Execution
+
+A continuous query uses a listener that is notified when:
+
+* An entry starts matching the specified query, represented by a _Join_ event.
+* A matching entry is updated and continues to match the query, represented by an _Update_ event.
+* An entry stops matching the query, represented by a _Leave_ event.
+
+When a client registers a continuous query listener it immediately begins to receive the results currently matching the
+query, received as _Join_ events as described above. In addition, it will receive subsequent notifications when other
+entries begin matching the query, as _Join_ events, or stop matching the query, as _Leave_ events, as a consequence of
+any cache operations that would normally generate creation, modification, removal, or expiration events. Updated cache
+entries will generate _Update_ events if the entry matches the query filter before and after the operation. To
+summarize, the logic used to determine if the listener receives a _Join_, _Update_ or _Leave_ event is:
+
+. If the query on both the old and new values evaluate false, then the event is suppressed.
+. If the query on the old value evaluates false and on the new value evaluates true, then a _Join_ event is sent.
+. If the query on both the old and new values evaluate true, then an _Update_ event is sent.
+. If the query on the old value evaluates true and on the new value evaluates false, then a _Leave_ event is sent.
+. If the query on the old value evaluates true and the entry is removed or expired, then a _Leave_ event is sent.
+
+NOTE: Continuous Queries can use the full power of the Query DSL except: grouping, aggregation, and sorting operations.
+
+=== Running Continuous Queries
+
+To create a continuous query you'll start by creating a Query object first. This is described in
```

Comment:
This seems like it could be broken into a set of steps. Maybe something like the following sequence:

To create a continuous query, do the following:

. Create a Query object. See link:#query_dsl[the Query DSL section].
. Obtain the ContinuousQuery (_org.infinispan.query.api.continuous.ContinuousQuery_) object of your cache by calling the appropriate method:
+
* _org.infinispan.client.hotrod.Search.getContinuousQuery(RemoteCache<K, V> cache)_ for {brandname} servers.
* _org.infinispan.query.Search.getContinuousQuery(Cache<K, V> cache)_ for embedded {brandname}.
+
. Register the query and a continuous query listener (_org.infinispan.query.api.continuous.ContinuousQueryListener_) as follows: